src/main/java/net/rptools/maptool/client/functions/MapFunctions.java

```diff
@@ -280,21 +282,22 @@ public Object childEvaluate(
 
       if (config.has("map asset")) {
         final var mapAssetId = config.getAsJsonPrimitive("map asset").getAsString();
-        final var mapAssetKey = FunctionUtil.getAssetKeyFromString(mapAssetId);
-        if (mapAssetKey == null) {
-          throw new ParserException(
-              I18N.getText("macro.function.map.invalidAsset", functionName, mapAssetId));
+        final var mapAssetKey = new AssetResolver().getAssetKey(mapAssetId);
+        MD5Key assetMD5 = null;
+        if (mapAssetKey.isPresent()) {
+          assetMD5 = new MD5Key(mapAssetKey.get().toString());
+        } else {
+          assetMD5 = new MD5Key(mapAssetId);
```

Comment:
I haven't tried it out, but this looks like it hides any failure when the user passes an invalid MD5. Or it would fail later on, but then the message would be less informative. We should prefer giving a clear and direct error message.

Reply:
I've reinstated the exception with the error